redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java

```diff
@@ -111,11 +136,26 @@ private void updateCloudPlatformAndRelatedFields(AllocateDatabaseServerV4Request
         dbStack.setPlatformVariant(cloudPlatform);
     }
 
-    private Network buildNetwork(NetworkV4Request source) {
+    private MappableBase getNetworkFromEnvironment(DetailedEnvironmentResponse environmentResponse) {
+        if (environmentResponse.getNetwork() == null || environmentResponse.getNetwork().getSubnetMetas() == null) {
+            throw new RedbeamsException("Environment does not contain metadata for subnets");
+        }
+        List<CloudSubnet> subnetMetas = new ArrayList<>(environmentResponse.getNetwork().getSubnetMetas().values());
+        List<String> chosenSubnetIds = subnetChooserService.chooseSubnetsFromDifferentAzs(subnetMetas).stream()
+                .map(CloudSubnet::getId)
+                .collect(Collectors.toList());
+
+        return networkParameterFactoryService.createNetworkParameters(chosenSubnetIds, environmentResponse.getCloudPlatform());
```

Comment:
The cloud platform is already available as an enum in the AllocateDatabaseServerV4Request object. It would be better to use that so that the service called here can switch on the enum instead of strings.

This does surface a problem - what happens when the cloud platform in the environment is different from the cloud platform in the request? That's something to think about as a separate item.

Reply:
That was exactly my concern as well and that's why I did go with the "inferior" string solution instead of a nice enum. If we digest what the env returns we are safe.

Yes, I agree, we should check if the providers do match in the request and the actual env. After that, however, we could use the enums.

Reply:
I created this jira to follow up:
https://jira.cloudera.com/browse/CB-2425